packages/react-native/React/Views/RCTCursor.m

```diff
@@ -0,0 +1,233 @@
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// [macOS]
+
+#import <Foundation/Foundation.h>
+#import "RCTCursor.h"
+
+#if defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 150000 /* __MAC_15_0 */
+#define __MAC_OS_15_SDK_AVAILABLE
+#endif // __MAC_OS_X_VERSION_MAX_ALLOWED
+
+#if TARGET_OS_OSX
+NSCursor *NSCursorFromRCTCursor(RCTCursor cursor)
+{
+  NSCursor *resolvedCursor = nil;
+  switch (cursor) {
+    case RCTCursorAuto:
+      break;
+    case RCTCursorAlias:
+      resolvedCursor = [NSCursor dragLinkCursor];
+      break;
+    case RCTCursorAllScroll:
+      // Not supported
+      break;
+    case RCTCursorCell:
+      // Not supported
+      break;
+    case RCTCursorColResize:
+#ifdef __MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
+        resolvedCursor = [NSCursor columnResizeCursor];
+      } else {
+        resolvedCursor = [NSCursor resizeLeftRightCursor];
+      }
+#else
+      resolvedCursor = [NSCursor resizeLeftRightCursor];
+#endif // __MAC_OS_15_SDK_AVAILABLE
+      break;
+    case RCTCursorContextMenu:
+      resolvedCursor = [NSCursor contextualMenuCursor];
+      break;
+    case RCTCursorCopy:
+      resolvedCursor = [NSCursor dragCopyCursor];
+      break;
+    case RCTCursorCrosshair:
+      resolvedCursor = [NSCursor crosshairCursor];
+      break;
+    case RCTCursorDefault:
+      resolvedCursor = [NSCursor arrowCursor];
+      break;
+    case RCTCursorEResize:
+#ifdef __MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
+        resolvedCursor = [NSCursor frameResizeCursorFromPosition:NSCursorFrameResizePositionLeft
+                                                    inDirections:NSCursorFrameResizeDirectionsOutward];
+      } else {
+        resolvedCursor = [NSCursor resizeRightCursor];
+      }
+#else
+      resolvedCursor = [NSCursor resizeRightCursor];
+#endif // __MAC_OS_15_SDK_AVAILABLE
+      break;
+    case RCTCursorEWResize:
+#ifdef __MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
+        resolvedCursor = [NSCursor frameResizeCursorFromPosition:NSCursorFrameResizePositionLeft
+                                                    inDirections:NSCursorFrameResizeDirectionsAll];
+      }
+#endif // __MAC_OS_15_SDK_AVAILABLE
+      break;
+    case RCTCursorGrab:
+      resolvedCursor = [NSCursor openHandCursor];
+      break;
+    case RCTCursorGrabbing:
+      resolvedCursor = [NSCursor closedHandCursor];
+      break;
+    case RCTCursorHelp:
+      // Not supported
+      break;
+    case RCTCursorMove:
+      // Not supported
+      break;
+    case RCTCursorNEResize:
+#ifdef __MAC_OS_15_SDK_AVAILABLE
+      if (@available(macOS 15.0, *)) {
+        resolvedCursor = [NSCursor frameResizeCursorFromPosition:NSCursorFrameResizePositionTopRight
+                                                    inDirections:NSCursorFrameResizeDirectionsOutward];
+      }
+      break;
+#endif // __MAC_OS_15_SDK_AVAILABLE
```

Comment:
Is this intentional? Is `RCTCursorNEResize` supposed to fallthrough to the case below?

Reply:
AKA, the break should be outside the ifdef? Good catch.